packages/fxa-auth-server/lib/payments/configuration/product.ts

```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import joi from '@hapi/joi';
+
+import {
+  BaseConfig,
+  baseConfigSchema,
+  CapabilityConfig,
+  StyleConfig,
+  SupportConfig,
+  UiContentConfig,
+  UrlConfig,
+} from './base';
+
+export const productConfigSchema = baseConfigSchema
+  .keys({
+    stripeProductId: joi.string().optional(),
```

Comment:
At the moment, our front-end logic requests this information using a Stripe product/plan id, which means any ProductConfig or PlanConfig without it will not be returned. So not having this cannot affect the front-end, since there's no way for the front-end to retrieve it in that case.

Reply:
Yes that's true. But functionally, if we don't require these IDs, then there'd be no indication say to a PM entering a new product or plan into the future config UI that this information is necessary for them to see their product/plan on our payments website.

They'll get an opaque error and ping us, and after digging into it, we'll find out the product/plan they want isn't being listed, and we'll have to figure out why and do a manual validation check (which is our current situation).

It seems like it defeats one of the benefits of moving this stuff into Firestore -- validating before the data is stored/reflected back?

Reply:
Validation will never be a 100% thing unfortunately, and I don't think we should strive for that. Consider that we could very easily put a large UX warning at the top saying, "THIS PLAN NOT CONFIGURED FOR WEB UNTIL STRIPE PLAN IS SELECTED." at the top. If they're making a apple/google only plan, they can see and ignore that, but there's easy visual ways we can clearly indicate what this means.